### ISSUE TITLE
After upgrading to ubuntu 20.04 , adsb-box.lighttpd fails to start

### ISSUE THREAD

**Issue by ldti** [tsunghanliu/adsb-box.snap#28]
log only seems to include the following:

```2020-05-03T17:10:30Z systemd[1]: Stopped Service for snap application adsb-box.lighttpd.
2020-05-03T17:10:30Z systemd[1]: Started Service for snap application adsb-box.lighttpd.
2020-05-03T17:10:30Z adsb-box.lighttpd[7083]: 2020-05-03 20:10:30: (server.c.738) couldn't get 'max filedescriptors' Operation not permitted
2020-05-03T17:10:30Z systemd[1]: snap.adsb-box.lighttpd.service: Main process exited, code=exited, status=255/EXCEPTION
2020-05-03T17:10:30Z systemd[1]: snap.adsb-box.lighttpd.service: Failed with result 'exit-code'.
2020-05-03T17:10:30Z systemd[1]: snap.adsb-box.lighttpd.service: Scheduled restart job, restart counter is at 5.
2020-05-03T17:10:30Z systemd[1]: Stopped Service for snap application adsb-box.lighttpd.
2020-05-03T17:10:30Z systemd[1]: snap.adsb-box.lighttpd.service: Start request repeated too quickly.
2020-05-03T17:10:30Z systemd[1]: snap.adsb-box.lighttpd.service: Failed with result 'exit-code'.
2020-05-03T17:10:30Z systemd[1]: Failed to start Service for snap application adsb-box.lighttpd.```

**Comment by tsunghanliu** [comment]
I did a fresh installation, but didn't have this issue. It seems the interfaces of the adsb-box snap aren't connected properly.
Could you please proved some logs for debugging?
`sudo journalctl -b | grep adsb-box`, `snap connections adsb-box`

**Comment by ldti** [comment]
I will try to reproduce this on a vm, as upgrading the PI every time is very time-consuming…

**Comment by ldti** [comment]
Unfortunately, I was unable to reproduce the issue on a vm (installing 19.10 then upgrading to 20.04).
I will try later today to upgrade the PI again, and perhaps reinstall the snap.

**Comment by ldti** [comment]
and the connections:

```root@ubuntu:~# snap connections adsb-box
Interface         Plug                       Slot               Notes
hardware-observe  adsb-box:hardware-observe  :hardware-observe  manual
mount-observe     adsb-box:mount-observe     :mount-observe     manual
network           adsb-box:network           :network           -
network-bind      adsb-box:network-bind      :network-bind      -
network-observe   adsb-box:network-observe   :network-observe   manual
process-control   adsb-box:process-control   :process-control   manual
raw-usb           adsb-box:raw-usb           :raw-usb           manual
system-observe    adsb-box:system-observe    :system-observe    manual ```

**Comment by tsunghanliu** [comment]
The journal log shows:

```
May 04 18:17:12 ubuntu audit[5881]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=5881 comm="lighttpd" exe="/snap/adsb-box/259/usr/sbin/lighttpd" sig=0 arch=c00000b7 syscall=163 compat=0 ip=0xffffb092ad08 code=0x50000
May 04 18:17:12 ubuntu audit[5881]: SECCOMP auid=4294967295 uid=0 gid=0 ses=4294967295 pid=5881 comm="lighttpd" exe="/snap/adsb-box/259/usr/sbin/lighttpd" sig=0 arch=c00000b7 syscall=163 compat=0 ip=0xffffb092ad08 code=0x50000
May 04 18:17:12 ubuntu adsb-box.lighttpd[5881]: 2020-05-04 18:17:12: (server.c.738) couldn't get 'max filedescriptors' Operation not permitted
May 04 18:17:12 ubuntu systemd[1]: snap.adsb-box.lighttpd.service: Main process exited, code=exited, status=255/EXCEPTION
May 04 18:17:12 ubuntu systemd[1]: snap.adsb-box.lighttpd.service: Failed with result 'exit-code'.
```

This means the syscall 163 (getrlimit) is blocked by seccomp. I'll ask the snapd team if they changed the policy or there's something new on Ubuntu 20.04.

**Comment by ldti** [comment]
Thank you!



Sent from my Samsung Galaxy smartphone.

**Comment by tsunghanliu** [comment]
I've raised the [question](https://forum.snapcraft.io/t/getrlimit-blocked-by-seccomp-on-focal-arm64/17237) on the forum. But, I didn't get a chance to verify the content of `/var/lib/snapd/seccomp/snap.adsb-box.lighttpd.src` yet.

@ldti are you convenient to run `grep getrlimit /var/lib/snapd/seccomp/snap.adsb-box.lighttpd.src` on the device? Also, did you ever try a fresh install of Ubuntu 20.04?

**Comment by ldti** [comment]
```
root@ubuntu:/var/lib/snapd# grep getrlimit /var/lib/snapd/seccomp/bpf/snap.adsb-box.lighttpd.src
getrlimit
ugetrlimit
```
a fresh install seems to be working fine (in the lighttpd sense, didn't do a full check with rtl-sdr), though it was tested on a x64 vm, not rpi.

**Comment by satmandu** [comment]
I'm seeing the same problem on device upgraded to 20.04.
```
uname -a
Linux rpi4 5.4.40-v8+ #1316 SMP PREEMPT Tue May 12 13:17:34 BST 2020 aarch64 aarch64 aarch64 GNU/Linux
$ snap services adsb-box
Service                       Startup  Current   Notes
adsb-box.adsbexchange-mlat    enabled  active    -
adsb-box.adsbexchange-netcat  enabled  active    -
adsb-box.collectd             enabled  active    -
adsb-box.dump1090             enabled  active    -
adsb-box.fr24feed             enabled  active    -
adsb-box.graphs-gend          enabled  active    -
adsb-box.lighttpd             enabled  inactive  -
adsb-box.openskyd             enabled  active    -
adsb-box.pfclient             enabled  active    -
adsb-box.piaware              enabled  active    -
adsb-box.rbfeeder             enabled  active    -
adsb-box.rbfeeder-mlat        enabled  active    -

satadru@rpi4:~$ sudo snap connections adsb-box
Interface         Plug                       Slot               Notes
hardware-observe  adsb-box:hardware-observe  :hardware-observe  manual
mount-observe     adsb-box:mount-observe     :mount-observe     manual
network           adsb-box:network           :network           -
network-bind      adsb-box:network-bind      :network-bind      -
network-observe   adsb-box:network-observe   :network-observe   manual
process-control   adsb-box:process-control   :process-control   manual
raw-usb           adsb-box:raw-usb           :raw-usb           manual
system-observe    adsb-box:system-observe    :system-observe    manual
```
```
sudo journalctl -b | grep adsb-box | pastebinit
```
[https://paste.ubuntu.com/p/SPYS8s6p86/](https://paste.ubuntu.com/p/SPYS8s6p86/)

**Comment by tsunghanliu** [comment]
The issue is caused by libseccomp.
In short, some syscall numbers are not exactly the same with kernel 5.4+, The libseccomp maintainers and Ubuntu maintainers are working on the fix actively.

To workaround this, either staying with the old Ubuntu releases or reinstalling adsb-box with `--devmode` should work.

**Comment by ldti** [comment]
OK, this helped. now I am encountering another issue with fr24feed. I will open it separately.

**Comment by ldti** [comment]
Cancel the additional issue. a restart solved it, for some reason.

**Comment by tsunghanliu** [comment]
This issue should be fixed by a newer version of snapd months ago.

Going to close this issue.
